configure.ac

```diff
@@ -121,6 +123,9 @@ AS_IF([test "${NCCL_OFI_PLATFORM}" = "none"], [AX_CHECK_PLATFORM_AWS()])
 
 AC_SUBST([NCCL_NET_OFI_DISTCHCK_CONFIGURE_FLAGS])
 
+AS_IF([test -n "${with_valgrind}" -a "${with_valgrind}" != "no" -a "${enable_asan}" = "yes"],
```

Comment:
This is not the exact same valgrind test you used in the valgrind configure.  Either add arguments to the valgrind macro to figure out if its enabled or not, or leak a variable that tells you if it is enabled or not.  But this is overly confusing for no reason.

Reply:
I would vote for the argument approach. I assume you mean that we pass the argument name as parameter and then access it via `$1` and `${$1}`. This boils down to a variable leakage as well I think but makes it more obvious in the `configure.ac` file.

Reply:
Normally, you would have two arguments, the one evaluated if the package is found and the one that is evaluated if the package is not found.

Reply:
Autoconf style is that instead of:
```
CHECK_VALGRIND([valgrind_enabled])
```
and then having `CHECK_VALGRIND` set `valgrind_enabled` to 0 or 1, you instead do:
```
CHECK_VALGRIND([valgrind_enabled=1], [valgrind_enabled=0])
```
and the `CHECK_VALGRIND` macro has a line like this:
```
AS_IF([test "$my_happy_check" = "yes"], [$1], [$2])
```

Reply:
Thank you.